### PR TITLE
initial implementation of the code inspector lint

### PR DESCRIPTION
**Pull request by oli-obk**
cc #1234

I left a lot of stuff commented out, since I want to implement it but haven't yet. Hope that's alright with everyone.

### CODE REVIEW

File: clippy_lints/src/utils/inspector.rs [hunk 1]
```diff
@@ -0,0 +1,179 @@
+//! checks for attributes
+
+use rustc::lint::*;
+use rustc::hir;
+use syntax::ast::{Attribute, MetaItemKind};
+
+/// **What it does:** Dumps every ast/hir node which has the `#[inspect]` attribute
```
Comment (mcarton) [inline clippy_lints/src/utils/inspector.rs]: Maybe you should choose a less generic name.

File: clippy_lints/src/utils/inspector.rs [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+///
+/// **Why is this bad?** 😈
```
Comment (mcarton) [inline clippy_lints/src/utils/inspector.rs]: I don't have that character in my font :disappointed:

File: clippy_lints/src/utils/inspector.rs [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+
+/// **What it does:** Dumps every ast/hir node which has the `#[clippy_dump]` attribute
+///
+/// **Why is this bad?** 😈
+///
+/// **Known problems:** ∅
+///
+/// **Example:**
+/// ```rust
+/// #[inspect]
```
Comment (mcarton) [inline clippy_lints/src/utils/inspector.rs]: Is it `inspect` or `clippy-dump`?

File: clippy_lints/src/utils/inspector.rs [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+
+//! checks for attributes
+
+use rustc::lint::*;
+use rustc::hir;
+use syntax::ast::{Attribute, MetaItemKind};
+
+/// **What it does:** Dumps every ast/hir node which has the `#[clippy_dump]` attribute
+///
+/// **Why is this bad?** 😈
```
Comment (mcarton) [inline clippy_lints/src/utils/inspector.rs]: That's not very helpful :confused:

Comment (oli-obk) [inline clippy_lints/src/utils/inspector.rs]: not sure what else to do here? remove that question and deviate from the usual doc pattern?

Comment (mcarton) [inline clippy_lints/src/utils/inspector.rs]: Yup, this is not the usual kind of lint

**Pull request merged**